bahmni_stock/models/stock_picking.py

```diff
@@ -1,20 +1,27 @@
 # -*- coding: utf-8 -*-
 from collections import namedtuple
 
-from odoo import fields, models, api
+from odoo import fields, models, api, _
 from odoo.tools.float_utils import float_compare
 from odoo.exceptions import UserError
 from odoo.exceptions import UserError, ValidationError, AccessError, RedirectWarning
 
+import logging
+
+_logger = logging.getLogger(__name__)
+
 
 class StockPicking(models.Model):
     _inherit = 'stock.picking'
 
     def button_validate(self):
         self.validate_batch_quantity_for_internal_transfer()
-        res = super(StockPicking, self).button_validate()
-        self.update_price_details_for_lots()
-        return res
+        if self._is_validation_needed():
+            return self.show_validation_screen()
+        else:
+            res = super(StockPicking, self).button_validate()
+            self.update_price_details_for_lots()
+            return res
 
     def validate_batch_quantity_for_internal_transfer(self):
         if self.picking_type_id and self.picking_type_id.code == 'internal':
```

Comment:
Better we move these to constant file and refer here if possible or may be declare in same file and refer here as 
(self.PICKING_TYPE_INTERNAL) for all the hardcoded strings like 'incoming' and 'Receipts' aswell

Reply:
Refactored @deepthi-mantena. Thanks